test/unit/data/test_galaxy_mapping.py

```diff
@@ -154,6 +155,36 @@ def assert_display_name_converts_to_unicode(item, name):
         assert isinstance(history.get_display_name(), text_type)
         assert history.get_display_name() == u'Hello₩◎ґʟⅾ'
 
+    def test_hda_to_library_dataset_dataset_association(self):
+        u = self.model.User(email="mary@example.com", password="password")
+        hda = self.model.HistoryDatasetAssociation(name='hda_name')
+        self.persist(hda)
+        trans = collections.namedtuple('trans', 'user')
+        target_folder = self.model.LibraryFolder(name='library_folder')
+        ldda = hda.to_library_dataset_dataset_association(
+            trans=trans(user=u),
+            target_folder=target_folder,
+        )
+        assert target_folder.item_count == 1
+        assert ldda.id
+        assert ldda.library_dataset.id
+        assert ldda.library_dataset_id
```

Comment:
```suggestion
```

Reply:
I've done this intentionally. One of my earlier attempts was a backref only that wouldn't populate `library_dataset_id `

Reply:
My eyes saw 2 duplicate lines, sorry!

Reply:
That has to be a first time for 🦅 👀 ! 😆